### PR TITLE
[SYCL][ESIMD] Add support for esimd_trunc function

### PR DESCRIPTION
**Pull request by sndmitriev**
Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 1]
```diff
@@ -1914,6 +1914,17 @@ ESIMD_INLINE RT esimd_ceil(const float &src0, const uint flags = 0) {
   return esimd_rndu<RT, 1U>(src0, flags);
 }
 
+template <typename RT, int SZ>
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1917]: please add doxygen comments

**Comment by kbobrovs** [comment]
@sndmitriev , this is first part, which will be followed by changes in LowerESIMD.cpp right?
Please also add a E2E test to https://github.com/intel/llvm-test-suite/tree/intel/SYCL/ESIMD and add a link here.

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp (hunk 1 above)
Comment (sndmitriev) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1925]: Done.

**Comment by sndmitriev** [comment]
> @sndmitriev, please also add a E2E test to https://github.com/intel/llvm-test-suite/tree/intel/SYCL/ESIMD and add a link here.
> 
> (removed my other note about LowerESIMD.cpp, as no changes are needed there)

Link to the test suite's PR - https://github.com/intel/llvm-test-suite/pull/477

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**